pykern/util.py

```diff
@@ -65,3 +66,51 @@ def _check_dev_files(root):
         # Don't run from an install directory
         r = pkio.py_path()
     return pkio.mkdir_parent(r.join(_DEFAULT_ROOT))
+
+
+def is_pure_text(value, test_size=512):
+    """Guesses if value is pure text using heuristics.
+
+    Args:
+        value (bytes): bytes data
+        test_size (int): size of bytes chunk being tested
```

Comment:
I don't think `test_size` is relevant here. You have `value` and `len(value)` is what you are interested in. The algorithm should backup a few bytes to see if it can pass utf8 decoding.

The only problem you have here is if the value passed from pkio  is shorter than its test_size. That's a tricky relationship. I think what you really want is a flag "maybe_truncated" that pkio passes. If it is not truncated, then you don't backup, because you have the whole string here. The is_truncated flag is set in pkio by reading test_size + 1 and then checking if the read value is test_size + 1. If it isn't, then it isn't truncated. You only pass `value[:test_size]`, i.e. `util.is_pure_text(b[:test_size], is_truncated=len(b) > test_size)`.